src/utils/markdown_processor.py

```
@@ -0,0 +1,948 @@
+        if rule.get("required", False) and field not in frontmatter:
+            errors.append(f"Required field '{field}' is missing")
+
+    return {
+        "is_valid": len(errors) == 0,
+        "errors": errors,
+        "normalized_frontmatter": normalized,
+        "validation_timestamp": datetime.now().isoformat(),
+    }
+
+
+def _normalize_field_value(field: str, value: str, allowed_values: list[str]) -> str | None:
```

Pylint (3.3.7)

[warning] Unused argument 'allowed_values'

(W0613)